crates/types/src/invocation.rs

```diff
@@ -1247,3 +1369,67 @@ mod mocks {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    mod invocation_response {
+        use super::*;
+
+        #[allow(non_camel_case_types)]
+        #[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+        pub struct InvocationResponse_1_2 {
+            pub id: InvocationId,
+            pub entry_index: EntryIndex,
+            pub result: ResponseResult,
+        }
+
+        #[test]
+        fn front_compat() {
+            let caller_id = InvocationId::mock_random();
+            let caller_completion_id = 10;
+            let result = ResponseResult::Success(Bytes::from_static(b"123"));
+
+            let old_response = InvocationResponse_1_2 {
+                id: caller_id,
+                entry_index: caller_completion_id,
+                result: result.clone(),
+            };
+
+            let new_response: InvocationResponse =
+                serde_json::from_str(&serde_json::to_string(&old_response).unwrap()).unwrap();
+
+            assert_eq!(new_response.target.caller_id, caller_id);
+            assert_eq!(
+                new_response.target.caller_completion_id,
+                caller_completion_id
+            );
+            assert_eq!(new_response.target.caller_invocation_epoch, 0);
+            assert_eq!(new_response.result, result);
+        }
+
+        #[test]
+        fn back_compat() {
```

Comment:
This looks like the forward compatibility test: The old type can read the data serialized by the new version/type.